apps/mobile/src/features/token/utils/render-collectible-details.tsx

```diff
@@ -0,0 +1,120 @@
+import { truncateMiddle } from '@leather.io/utils';
+
+import { TokenDetailsCard } from '../components/token-details-card';
+
+function formatLabel(key: string): string {
+  // Special case for locationUrl in collection context
+  if (key === 'location_url') {
+    return t`Collection URL`;
+  }
+
+  // Replace underscores with spaces
+  let formatted = key.replace(/_/g, ' ');
+
+  // Add spaces between PascalCase words
+  formatted = formatted.replace(/([a-z])([A-Z])/g, '$1 $2');
+
+  // Capitalize first letter
+  return formatted.charAt(0).toUpperCase() + formatted.slice(1);
+}
+
+function renderValue(value: any, key: string): React.ReactNode {
+  const stringValue = Array.isArray(value) ? JSON.stringify(value) : String(value);
+
+  const keysToTruncate = ['address', 'id', 'assetId', 'contractId', 'txid'];
+  if (keysToTruncate.includes(key)) {
+    return truncateMiddle(stringValue, 4);
+  }
+
+  // Special handling for location_url - prepend gamma.io domain
+  if (key === 'location_url' && typeof stringValue === 'string') {
+    const fullUrl = `https://gamma.io${stringValue}`;
+    return (
+      <Pressable onPress={() => Linking.openURL(fullUrl)}>
+        {({ pressed }) => (
+          <Text textDecorationLine="underline" opacity={pressed ? 0.5 : 1}>
+            {t`View`}
+          </Text>
+        )}
+      </Pressable>
+    );
+  }
+
+  // If value is a URL, make it clickable
+  if (typeof stringValue === 'string' && stringValue.startsWith('https://')) {
+    return (
+      <Pressable onPress={() => Linking.openURL(stringValue)}>
+        {({ pressed }) => (
+          <Text textDecorationLine="underline" opacity={pressed ? 0.5 : 1}>
+            {t`View`}
+          </Text>
+        )}
+      </Pressable>
+    );
+  }
+
+  return stringValue;
+}
+
+function getSortedEntries(obj: Record<string, any>, sortOrder: string[]) {
+  return Object.entries(obj).sort(([keyA], [keyB]) => {
+    const indexA = sortOrder.indexOf(keyA);
+    const indexB = sortOrder.indexOf(keyB);
+
+    // If both are in sortOrder, sort by their position
+    if (indexA !== -1 && indexB !== -1) return indexA - indexB;
+    // If only A is in sortOrder, A comes first
+    if (indexA !== -1) return -1;
+    // If only B is in sortOrder, B comes first
+    if (indexB !== -1) return 1;
+    // Otherwise maintain original order
+    return 0;
+  });
+}
+
+export function renderCollectibleDetailsRecursively(
+  obj: Record<string, any>,
+  title?: string
+): React.ReactNode[] {
+  const items: React.ReactNode[] = [];
+  const nestedCards: React.ReactNode[] = [];
+
+  // Defined sort order for the collectible details
+  const sortOrder = ['name', 'tokenId', 'assetId', 'contractId', 'mimetype', 'contentType'];
+  getSortedEntries(obj, sortOrder).forEach(([key, value]) => {
+    // Skip undefined or empty string values
+    if (value === undefined || value === '') return;
+
+    const keysToSkip = ['description', 'floorPrice', 'floor_price_amount'];
```

Comment:
@alexp3y I left out `floor_price` for now. We were discussing it [here](https://github.com/leather-io/mono/pull/1635#discussion_r2362716204). If we really want it I can convert it to `Money` but for now I've just filtered it out